src/shared_modules/router/tests/component/interface_test.cpp

```diff
@@ -37,6 +37,22 @@ TEST_F(RouterInterfaceTest, TestCreateProviderSubscriberSimple)
     });
 }
 
+TEST_F(RouterInterfaceTest, TestStopLocalProviderWithoutStart)
+{
+    auto provider = std::make_unique<RouterProvider>("test");
+
+    // Simulate call from specific provider in the same process
+    EXPECT_THROW(provider->stop(), std::runtime_error);
+}
+
+TEST_F(RouterInterfaceTest, TestCreateSubscriberWithoutProvider)
+{
+    EXPECT_NO_THROW({
+        auto subscriptor = std::make_unique<RouterSubscriber>("test", "subscriberTest");
+        subscriptor->subscribe([](const std::vector<char>&) {});
```

Comment:
Shouldn't this be ... ?
```suggestion
    EXPECT_NO_THROW({
        auto subscriber = std::make_unique<RouterSubscriber>("test", "subscriberTest");
        subscriber->subscribe([](const std::vector<char>&) {});
```

Reply:
same as above

Reply:
done